From 0c6db3f2118dbb723529b24af11a136d19900244 Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Fri, 27 Jan 2012 14:35:01 +0100
Subject: [PATCH 3/8] Add LANGUAGE_SUPPORT what-provides type

This is useful to ask "which packages to I need to get full support for a given
language/locale", i. e. wordlists, dictionaries, translated help packages, ibus
modules, and the like.
---
 docs/provides-component-naming.txt |   12 ++++++++++++
 lib/packagekit-glib2/pk-enum.c     |    1 +
 lib/packagekit-glib2/pk-enum.h     |    1 +
 lib/packagekit-qt2/transaction.h   |    3 ++-
 4 files changed, 16 insertions(+), 1 deletions(-)

Index: packagekit-0.7.2/lib/packagekit-glib2/pk-enum.c
===================================================================
--- packagekit-0.7.2.orig/lib/packagekit-glib2/pk-enum.c	2012-01-17 17:22:30.000000000 +0100
+++ packagekit-0.7.2/lib/packagekit-glib2/pk-enum.c	2012-01-30 11:06:48.154839621 +0100
@@ -366,6 +366,7 @@
 	{PK_PROVIDES_ENUM_PLASMA_SERVICE,	"plasma-service"},
 	{PK_PROVIDES_ENUM_SHARED_LIB,		"shared-library"},
 	{PK_PROVIDES_ENUM_PYTHON,		"python-module"},
+	{PK_PROVIDES_ENUM_LANGUAGE_SUPPORT,     "language-support"},
 	{0, NULL}
 };
 
Index: packagekit-0.7.2/lib/packagekit-glib2/pk-enum.h
===================================================================
--- packagekit-0.7.2.orig/lib/packagekit-glib2/pk-enum.h	2011-12-25 22:38:49.000000000 +0100
+++ packagekit-0.7.2/lib/packagekit-glib2/pk-enum.h	2012-01-30 11:06:48.154839621 +0100
@@ -468,6 +468,7 @@
 	PK_PROVIDES_ENUM_PLASMA_SERVICE,
 	PK_PROVIDES_ENUM_SHARED_LIB,
 	PK_PROVIDES_ENUM_PYTHON,
+	PK_PROVIDES_ENUM_LANGUAGE_SUPPORT,
 	PK_PROVIDES_ENUM_LAST
 } PkProvidesEnum;
 
Index: packagekit-0.7.2/lib/packagekit-qt2/transaction.h
===================================================================
--- packagekit-0.7.2.orig/lib/packagekit-qt2/transaction.h	2011-08-04 12:42:46.000000000 +0200
+++ packagekit-0.7.2/lib/packagekit-qt2/transaction.h	2012-01-30 11:07:45.518842395 +0100
@@ -338,7 +338,10 @@
         ProvidesFont,
         ProvidesHardwareDriver,
         ProvidesPostscriptDriver,
-        ProvidesPlasmaService
+        ProvidesPlasmaService,
+        ProvidesSharedLib,
+        ProvidesPythonModule,
+        ProvidesLanguageSupport
     } Provides;
 
     /**
